src/components/navigation-item.tsx

```diff
@@ -27,15 +27,15 @@ const NavigationItem = ({ isDone, isActive, slug, title, onClick }: Props) => {
   });
 
   let className = 'side-nav__item ';
-  if (isDone) {
+  if (isRead) {
     className += 'side-nav__item--done';
   } else if (isActive) {
     className += 'side-nav__item--active';
   }
 
   return (
     <Link
-      ref={handleRef}
+      innerRef={handleRef}
```

Comment:
Question for my education: why is innerRef preferred over ref? (Some lib like styled-components v4 started to use ref instead)

Reply:
`ref` is definitely preferred. 

This is just a temporary workaround to pacify TS. It basically was throwing a TypeError. @ollelauribostrom had logged an issue in the gatsbyjs repo: https://github.com/gatsbyjs/gatsby/issues/12014.

Quoting @freiksenet,
>We haven't updated gatsby-link to use React>=16.4 API yet, so you can only use function refs through innerRef function. ref isn't even an allowed prop type in Gatsby link component. We definitely do want to support it though.

Reply:
Oh, looks like https://github.com/gatsbyjs/gatsby/pull/12174 was merged! We need to upgrade our package.

Reply:
Upgraded packages (it didn't resolve the TypeError yet somehow, but that's no reason to go backwards). Reverted `innerRef`.